node_modules/gh-rest/lib/rest/user.js

```diff
@@ -98,7 +99,8 @@ module.exports = function(client) {
             'emailPreference': opts.emailPreference,
             'isAdmin': opts.isAdmin,
             'recaptchaChallenge': opts.recaptchaChallenge,
-            'recaptchaResponse': opts.recaptchaResponse
+            'recaptchaResponse': opts.recaptchaResponse,
+            'termsAndConditions': opts.termsAndConditions
```

Comment:
We persist the timestamp of when the T&C change in the config for each application (`config.termsAndConditionsLastUpdate`).
We persist the timestamp of when a user last accepted the T&C on their user profile (`user.termsAndConditions`).

On each request we'll check whether the user has accepted the T&C since they last were updated. There's no need to scan the user table and invalidate records